packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemFileAdapter.kt

```diff
@@ -0,0 +1,106 @@
+package expo.modules.filesystem.next
+
+import android.content.Context
+import android.net.Uri
+import androidx.core.net.toUri
+import androidx.documentfile.provider.DocumentFile
+import java.io.File
+import java.net.URI
+
+class FileSystemFileAdapter(val context: Context, val uri: Uri) {
```

Comment:
```suggestion
interface FileAdapter { ... }
class SafFileAdapter(val context, val uri: Uri) : FileAdapter { ... }
class SystemFileAdapter(val context, val uri: Uri) : FileAdapter { ... }

fun createFileAdapter(val context, val uri: Uri): FileAdapter {
  return if (uri.scheme == "content") {
    SafFileAdapter(context, uri)
  } else {
    SystemFileAdapter(context, uri)
  }
}
```
What do you think about splitting FileAdapter into two classes? It will simplify logic.